src/App.tsx

```diff
@@ -1,26 +1,395 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { TodoList } from './components/TodoList';
+import { TodosFilter } from './components/TodoFilter';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import classNames from 'classnames';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { wait } from './utils/fetchClient';
+import { TodoItem } from './components/TodoItem';
+import { CSSTransition } from 'react-transition-group';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [input, setInput] = useState('');
+  const [filter, setFilter] = useState(Filter.All);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const allChecked = todos.every((todo: Todo) => todo.completed);
+  const [isActive, setIsActive] = useState(allChecked);
+  const [serverError, setServerError] = useState(false);
+  const titleField = useRef<HTMLInputElement>(null);
+  const completedTodos = todos.filter((todo: Todo) => todo.completed);
+  const activeTodos = todos.filter((todo: Todo) => !todo.completed);
+  const todosToToggle = todos.filter(todo => todo.completed === isActive);
+
+  useEffect(() => {
+    setLoading(true);
+    todoService
+      .getTodos()
+      .then(todosFromServer => {
+        wait(1000);
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setErrorMessage(Error.UnableToLoadAll);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (titleField.current) {
+      return titleField.current.focus();
+    }
+  }, [isSubmitting]);
+
+  useEffect(() => {
+    setIsActive(allChecked);
+  }, [allChecked]);
+
+  useEffect(() => {
+    let timeoutId = 0;
+
+    if (errorMessage) {
+      timeoutId = window.setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+
+    return () => clearTimeout(timeoutId);
+  }, [errorMessage]);
+
+  if (!todoService.USER_ID) {
     return <UserWarning />;
   }
 
+  const handleFilterChange = (newFilter: Filter) => {
```

Comment:
redundant function wrapper - just use `setFilter` everywhere instead of `handleFilterChange`